src/main/java/uk/gov/hmcts/dts/fact/services/CourtService.java

```diff
@@ -76,4 +77,15 @@ public Court getCourtBySlug(final String slug) {
             .orElse(emptyList());
     }
 
+    public List<CourtReferenceWithDistance> getNearestCourtsByPostcodeAndAreaOfLawSearch(final String postcode, final String areaOfLaw) {
+        return mapitService.getCoordinates(postcode)
+            .map(value -> courtRepository
+                .findNearest(value.getLat(), value.getLon())
+                .stream()
+                .filter(c -> c.getAreasOfLaw().stream().anyMatch(a -> areaOfLaw.equalsIgnoreCase(a.getName())))
```

Comment:
I'm sure this is how the previous search is done but in general I would prefer the filter (and also the limit) to be part of the SQL query.

Reply:
i noticed the other searches using this, i believe its fine due to there being different parameters for the search i.e. you can seach with postcode and aol, or just postcode. for the future tickets, i believe theres a case where you search postcode or local authorities. I am happy to so do it sql way but due to time, think this might be eaiser